x-pack/plugins/security/public/management/role_combo_box/role_combo_box.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiComboBox, EuiText } from '@elastic/eui';
+import { Role, isRoleDeprecated } from '../../../common/model';
+
+interface Props {
+  availableRoles: Role[];
+  selectedRoleNames: string[];
+  onChange: (selectedRoleNames: string[]) => void;
+  placeholder?: string;
+  isLoading?: boolean;
+  isDisabled?: boolean;
+}
+
+export const RoleComboBox = (props: Props) => {
```

Comment:
Should we add some tests for this?

Reply:
Yeah I debated that for much longer than I should have. I can add tests for this, sure